nav2_costmap_2d/include/nav2_costmap_2d/costmap_2d_ros.hpp

```diff
@@ -121,6 +121,28 @@ class Costmap2DROS : public nav2_util::LifecycleNode
    */
   nav2_util::CallbackReturn on_shutdown(const rclcpp_lifecycle::State & state) override;
 
+  /**
+   * @brief as a child-LifecycleNode :
+   * Costmap2DROS may be launched by another Lifecycle Node as a composed module
+   * If composed, its parents will handle the shutdown, which includes this module
+   */
+  void on_rcl_preshutdown() override
```

Comment:
Do you mean we should modify the constructor in humble to look like in main branch?

Reply:
I'm saying that you should directly `git cherry-pick` the `main` commit to the `humble` branch and push that to this PR :smile: That way they're exactly the same. In fact, it might be wise to close this PR in general and reopen one with only the cherry picked commit.

Otherwise, if you wait like ~3 weeks, I'll be running a Humble sync and I'll get to this myself :smile: